src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/FindLargestCrossSections.cpp

```diff
@@ -0,0 +1,112 @@
+#include "complex/DataStructure/Geometry/ImageGeom.hpp"
+#include "complex/Utilities/DataArrayUtilities.hpp"
+
+using namespace complex;
+
+// -----------------------------------------------------------------------------
+FindLargestCrossSections::FindLargestCrossSections(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel,
+                                                   FindLargestCrossSectionsInputValues* inputValues)
+: m_DataStructure(dataStructure)
+, m_InputValues(inputValues)
+, m_ShouldCancel(shouldCancel)
+, m_MessageHandler(mesgHandler)
+{
+}
+
+// -----------------------------------------------------------------------------
+FindLargestCrossSections::~FindLargestCrossSections() noexcept = default;
+
+// -----------------------------------------------------------------------------
+const std::atomic_bool& FindLargestCrossSections::getCancel()
+{
+  return m_ShouldCancel;
+}
+
+// -----------------------------------------------------------------------------
+Result<> FindLargestCrossSections::operator()()
+{
+  const auto& imageGeom = m_DataStructure.getDataRefAs<ImageGeom>(m_InputValues->ImageGeometryPath);
+  auto& featureIds = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeatureIdsArrayPath);
+  auto& largestCrossSections = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->LargestCrossSectionsArrayPath);
+  const usize numFeatures = largestCrossSections.getNumberOfTuples();
+
+  // Validate the largestCrossSections array is the proper size
+  auto validateResults = ValidateNumFeaturesInArray(m_DataStructure, m_InputValues->LargestCrossSectionsArrayPath, featureIds);
+  if(validateResults.invalid())
+  {
+    return validateResults;
+  }
+
+  std::vector<float32> featureCounts(numFeatures, 0.0f);
+
+  usize outPlane = 0, inPlane1 = 0, inPlane2 = 0;
+  float32 resScalar = 0.0f, area = 0.0f;
+  usize stride1 = 0, stride2 = 0, stride3 = 0;
+  usize iStride = 0, jStride = 0, kStride = 0;
+  usize point = 0, gNum = 0;
+
+  FloatVec3 spacing = imageGeom.getSpacing();
+
+  if(m_InputValues->Plane == 0)
+  {
+    outPlane = imageGeom.getNumZCells();
+    inPlane1 = imageGeom.getNumXCells();
+    inPlane2 = imageGeom.getNumYCells();
+
+    resScalar = spacing[0] * spacing[1];
+    stride1 = inPlane1 * inPlane2;
+    stride2 = 1;
+    stride3 = inPlane1;
+  }
+  if(m_InputValues->Plane == 1)
+  {
+    outPlane = imageGeom.getNumYCells();
+    inPlane1 = imageGeom.getNumXCells();
+    inPlane2 = imageGeom.getNumZCells();
+    resScalar = spacing[0] * spacing[2];
+    stride1 = inPlane1;
+    stride2 = 1;
+    stride3 = inPlane1 * inPlane2;
+  }
+  if(m_InputValues->Plane == 2)
+  {
+    outPlane = imageGeom.getNumXCells();
+    inPlane1 = imageGeom.getNumYCells();
+    inPlane2 = imageGeom.getNumZCells();
+    resScalar = spacing[1] * spacing[2];
+    stride1 = 1;
+    stride2 = inPlane1;
+    stride3 = inPlane1 * inPlane2;
+  }
```

Comment:
```suggestion
    outPlane = imageGeom.getNumZCells();
    inPlane1 = imageGeom.getNumXCells();
    inPlane2 = imageGeom.getNumYCells();

  if(m_InputValues->Plane == 0)
  {
    resScalar = spacing[0] * spacing[1];
    stride1 = inPlane1 * inPlane2;
    stride2 = 1;
    stride3 = inPlane1;
  }
  if(m_InputValues->Plane == 1)
  {
    resScalar = spacing[0] * spacing[2];
    stride1 = inPlane1;
    stride2 = 1;
    stride3 = inPlane1 * inPlane2;
  }
  if(m_InputValues->Plane == 2)
  {
    resScalar = spacing[1] * spacing[2];
    stride1 = 1;
    stride2 = inPlane1;
    stride3 = inPlane1 * inPlane2;
  }
```
If `outPlane`, `inPlane1`, and `inPlane2` are not being altered within each if statement just declare them outside the repeated ifs so it is not called repeatedly. If memory space is a concern nest them (and the if statements) inside a code block to reduce their time in scope.

Reply:
@nyoungbq They are each set to different x, y, or z cell number values based on which plane so I don't think it makes sense to move them outside of the if statements.

Reply:
> @nyoungbq They are each set to different x, y, or z cell number values based on which plane so I don't think it makes sense to move them outside of the if statements.

Oh ok makes sense sorry for the confusion.